docs/decisions/20240103-use-jwt-claims-for-agent-admin-auth.md

```diff
@@ -0,0 +1,155 @@
+# Use JWT claims for agent admin access control
+
+- Status: accepted
+- Deciders: Pat Losoponkul, Yurii Shynbuiev, David Poltorak, Shailesh Patil
+- Date: 2024-01-03
+- Tags: multitenancy, authorisation, authentication
+
+Technical Story: [Allow agent admin role to be authenticated by Keycloak JWT | https://input-output.atlassian.net/browse/ATL-6074]
+
+## Context and Problem Statement
+
+External IAM authentication and authorization are currently limited to tenants using the wallets.
+Administrators currently rely on a static API key configured in the agent.
+Enhancing the security of admin access requires extending support for admin authentication through an external IAM.
+This introduces a concept of role to the authentication and authorization process.
+
+The existing tenant authentication and authorization model relies on UMA (user-managed access)
+which defines resources, policies, and permissions.
+Tenants gain access to resources based on defined permissions, effectively controlling resource access.
+In addition to wallet usage, the agent also handles wallet management,
+a functionality utilized by both tenants and administrators.
+While administrators don't directly use the wallet, they oversee its management.
+Integrating an auth model to distinguish between admins and tenants presents a new challenge.
+
+- Where and how to define the role of admin and tenant?
+- What boundary the admin role should be scoped to?
+- How to support different deployment topologies? (e.g. multi-agent in a shared realm, but not share the admin users)
+- How does it interact with the resource scope in the UMA model?
+
+## Decision Drivers
+
+- Must not prevent us from using other IAM systems
+- Must not prevent us from supporting fine-grained tenant wallet access in the future
+- Must allow SSO configuration
```

Comment:
It might be removed. SSO is about `authentication`, but in this ADR, we are making the decision about the `authorization` model.
Let's focus on the `authorization` in this ADR only.

Reply:
fixed